File: CSharpCodeBase/components/character/weaponsequencer.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class weaponsequencer :component{
 var WeaponSequencerComponent = Class({__includes=Component, init=function(self, entity)
       Component.init(self, entity);
       this.index = 1;
       this.weapons = this.entity.config:Get("weaponSequence")  ||  {}
       this.entity.weaponSequencer = self;
 }})
 public void WeaponSequencerComponent:GetNextWeapon(){
   if(#this.weapons == 0  ){ 
     return null;
   }
   var result = this.index;
   this.index = this.index + 1;
   if(this.index > #this.weapons  ){ 
     this.index = 1;
   }
   return this.weapons[result];
 }
 Component}}

File: CSharpCodeBase/components/character/weaponsequencer.lua
local Class = require 'utils.hump.class'
local Component = require 'components.component'

local WeaponSequencerComponent = Class({__includes=Component, init=function(self, entity)
      Component.init(self, entity)
      self.index = 1
      self.weapons = self.entity.config:Get('weaponSequence') or {}
      self.entity.weaponSequencer = self
end})

function WeaponSequencerComponent:GetNextWeapon()
  if #self.weapons == 0 then
    return nil
  end
  local result = self.index
  self.index = self.index + 1
  if self.index > #self.weapons then
    self.index = 1
  end
  return self.weapons[result]
end

return WeaponSequencerComponent